### ISSUE TITLE
Add an option to specify hidden notifications format

### ISSUE THREAD

**Issue by emersion** [emersion/mako#29]
Should be able to set the format to display something like "8 more". Should also be able to hide it with an empty format.

**Comment by adrubesh** [comment]
To fix this issue, I was thinking about changing the `format_notification` function to `size_t format_notification(struct mako_state *state, struct mako_notification *notif, const char *format,
char *buf)` 

This would allow us to use it for formatting things by notification state as well (like hidden notifications): https://github.com/adrubesh/mako/blob/hidden-notification-format/notification.c#L188

This would also clean up the render function and allow something like: https://github.com/adrubesh/mako/blob/hidden-notification-format/render.c#L141

**Comment by emersion** [comment]
That's a good idea, but I'm not a fan of setting `notif` to NULL: some variables make only sense for notifications, some other make only sense for hidden count…

What we could do instead is a generic

```c
typedef const char *mako_format_func_t(char variable, void *data);
size_t format_notification(const char *format, char *buf, mako_format_func_t func, void *data);
```

that would take a callback to replace variables. I wonder if there's a simpler solution, let me know what you think!

**Comment by adrubesh** [comment]
I am a fan of the idea. I will mess around with that and see what I can do.

**Comment by emersion** [comment]
Fixed #35
